fuel-vm/src/interpreter/initialization.rs

```diff
@@ -30,11 +32,20 @@ where
     /// Initialize the VM with a given transaction
     fn init_inner(
         &mut self,
-        tx: Tx,
+        mut tx: Tx,
         initial_balances: InitialBalances,
         runtime_balances: RuntimeBalances,
         gas_limit: Word,
     ) -> Result<(), RuntimeError<S::DataError>> {
+        if let Some(s) = tx.as_script_mut() {
+            *s.receipts_root_mut() = Default::default();
+        }
+        tx.inputs_mut()
+            .iter_mut()
+            .for_each(fuel_tx::Input::prepare_init_execute);
+        tx.outputs_mut()
+            .iter_mut()
+            .for_each(fuel_tx::Output::prepare_init_execute);
```

Comment:
Adding `prepare_init_execute` to `ExecutableTransaction` would mean adding `field::ReceiptRoot` bound, which is not satisfied by the `Create` transaction. Although, I'm not sure why `Create` is an `ExecutableTransaction` since there's nothing to execute there.

Reply:
You don't need to define the implementation inside of the `ExecutableTransaction`. You can have a separate implementation for `Create` and `Script` transactions